lighthouse-core/config/budget.js

```diff
@@ -98,31 +154,31 @@ class Budget {
       /** @type {LH.Budget} */
       const budget = {};
 
-      const {resourceSizes, resourceCounts, timings, ...invalidRest} = b;
+      const {path, resourceSizes, resourceCounts, timings, ...invalidRest} = b;
       Budget.assertNoExcessProperties(invalidRest, 'Budget');
 
-      if (b.resourceSizes !== undefined) {
-        budget.resourceSizes = b.resourceSizes.map((r) => {
+      budget.path = Budget.validatePath(path);
```

Comment:
It doesn't. And I may be getting ahead of myself here, but I think budget input will be mutated soon.

(Specifically, in #8539 I think converting user budgets from KB to bytes should be handled in budget.js rather than the audit.)

Once it does, I personally find it more readable when everything uses the assignment pattern rather than mixing styles. TLDR; I feel like this style is more future-proof?

Reply:
gotcha thanks for the explanation that makes sense!  ⚠️ counter point ahead :)

> rather than mixing styles

unfortunately I feel like that might be inevitable when we `assertNoExcessProperties` the line above this 😕  

as a reader, I start looking for how this method changes `path` but it doesn't. given that we will definitely have some functions that will assert and some that will mutate. how do you feel about aligning the non-mutating validation logic on `assert*` without assignment (which also aligns with our other config validation logic) and `mutate`/`normalize`/`someOtherModificationVerb` with the assignment pattern for the ones that do? :)

Reply:
FWIW, I have an imminent PR that does a little more budget validation that will helpfully make this decision more complicated :/